File: src/testing/demo-runtime.ts
import { SimpleMemory } from '../memory.js';
import { createAgentCLI, type AgentCLI } from '../runtime.js';
import type { VFS } from '../vfs/interface.js';
import { NodeVFS } from '../vfs/node-vfs.js';
import { DemoFetch, DemoSearch } from './adapters.js';
import { DEMO_FETCH_RESOURCES, DEMO_SEARCH_DOCS, seedDemoMemory, seedDemoVfs } from './demo-fixtures.js';

export interface DemoRuntimeOptions {
  rootDir?: string;
  vfs?: VFS;
}

export const seedVfs = seedDemoVfs;
export const seedMemory = seedDemoMemory;

function createDemoAdapters(): { search: DemoSearch; fetch: DemoFetch } {
  return {
    search: new DemoSearch(DEMO_SEARCH_DOCS),
    fetch: new DemoFetch(DEMO_FETCH_RESOURCES),
  };
}

export async function buildDemoRuntime(options: DemoRuntimeOptions = {}): Promise<AgentCLI> {
  const vfs = options.vfs ?? new NodeVFS(options.rootDir ?? './agent_state');
  await seedVfs(vfs);

  const memory = new SimpleMemory();
  seedMemory(memory);

  return createAgentCLI({
    vfs,
    adapters: createDemoAdapters(),
    memory,
  });
}

File: python/src/onetool/testing/demo_runtime.py
from __future__ import annotations

from dataclasses import dataclass

from ..memory import SimpleMemory
from ..runtime import create_agent_cli
from ..types import ToolAdapters
from ..vfs.interface import VFS
from ..vfs.memory_vfs import MemoryVFS
from .adapters import DemoFetch, DemoSearch
from .demo_fixtures import DEMO_FETCH_RESOURCES, DEMO_SEARCH_DOCS, seed_demo_memory, seed_demo_vfs


@dataclass(frozen=True, slots=True)
class DemoRuntimeOptions:
    vfs: VFS | None = None


seed_vfs = seed_demo_vfs
seed_memory = seed_demo_memory


async def build_demo_runtime(
    options: DemoRuntimeOptions | None = None,
    **runtime_options: object,
):
    resolved = options or DemoRuntimeOptions()
    create_options = dict(runtime_options)
    vfs = create_options.get("vfs") or resolved.vfs or MemoryVFS()
    await seed_vfs(vfs)

    memory = create_options.get("memory") or SimpleMemory()
    seed_memory(memory)

    create_options.setdefault("vfs", vfs)
    create_options.setdefault(
        "adapters",
        ToolAdapters(
            search=DemoSearch(DEMO_SEARCH_DOCS),
            fetch=DemoFetch(DEMO_FETCH_RESOURCES),
        ),
    )
    create_options.setdefault("memory", memory)

    return await create_agent_cli(**create_options)
